runtime/src/accounts_db.rs

```diff
@@ -3566,6 +3574,9 @@ impl AccountsDb {
                         result = AccountsIndexScanResult::Unref;
                         dead += 1;
                     } else {
+                        if all_are_zero_lamports && stored_account.account.lamports() != 0 {
```

Comment:
We can do this without an if:
`all_are_zero_lamports &= (stored_account.account.lamports == 0)`

Reply:
done